heron/tmaster/src/cpp/manager/tmaster.cpp

```diff
@@ -106,6 +104,12 @@ TMaster::TMaster(const std::string& _zk_hostport, const std::string& _topology_n
   do_reassign_ = false;
 
   master_establish_attempts_ = 0;
+
+  // start servers to get the actual ports before `tmaster_location_` related operation
```

Comment:
One of the fundamental assumptions in tmaster when it is processing requests is that it is already established itself as the master. This is to prevent multiple tmasters trampling on each other. Thats the reason, we do StartServer only after TMaster Establishment is complete.
This change starts the TMasterServer which is now ready to accept connections, even before it is established as master. Thus all sorts of bad issues can happen